gradle-plugin/src/gradleTest/java/com/squareup/anvil/plugin/LifecycleTest.kt

```diff
@@ -0,0 +1,90 @@
+package com.squareup.anvil.plugin
+
+import com.rickbusarow.kase.gradle.dsl.buildFile
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.string.shouldInclude
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.jupiter.api.TestFactory
+
+class LifecycleTest : BaseGradleTest() {
+
+  @TestFactory
+  fun `tasks are compatible with configuration caching when targeting kotlin-jvm`() =
+    kases.distinctBy { it.gradleVersion }
+      .asTests {
+
+        rootProject {
+          buildFile {
+            plugins {
+              kotlin("jvm", it.kotlinVersion)
+              id("com.squareup.anvil", anvilVersion)
+            }
+            dependencies {
+              compileOnly(libs.inject)
+            }
+          }
+
+          dir("src/main/java") {
+            injectClass()
+          }
+        }
+
+        val expected = when {
+          it.gradleVersion >= "8.5" -> "Calculating task graph as no cached configuration is available for tasks: compileKotlin"
+          it.gradleVersion >= "8.4" -> "Calculating task graph as no configuration cache is available for tasks: compileKotlin"
```

Comment:
These read to me like the messages that would be displayed at the start when attempting to store the configuration cache rather than confirming that it was successful / compatible. Is there another message that we can/should be checking here similar to the `else` case?

Reply:
Nice catch!  You're right.  I was conflating two different things here.

The assertions have been updated.